Python/chapter02/2.8 - Loop Detection/miguel_soln_2.8.py

```diff
@@ -164,6 +164,31 @@ def __eq__(self, other: object):
         return a is None and b is None
 
 
+def loop_detection_const_space(ll: LinkedList) -> Optional[Node]:
+    """
+    This function will determine if there is a
+    cycle in the input linked list.
+    A linked list is 'circular' when a node's
+    next pointer points to an earlier node, so
+    as to make a loop in the linked list.
+    Runtime:  O(n^2)
+    Space Complexity:  O(1)
+    :param ll: an input linked list
+    :return: the corrupt node or None
+    """
+    if ll.head and ll.head.next is ll.head:
```

Comment:
there doesn't seem to be a test case for this.

Reply:
that is for a single node linked list that has a next pointer pointing to itself

Reply:
The test method for that is `def test_loop_detection_single_node_ll`

Reply:
ah somehow i missed that. sorry about the noise.

Reply:
no worries